mongo/integration/unified/entity.go

```diff
@@ -48,34 +56,55 @@ type entityOptions struct {
 	DatabaseID string `bson:"database"`
 }
 
-// entityMap is used to store entities during tests. This type enforces uniqueness so no two entities can have the same
+// EntityMap is used to store entities during tests. This type enforces uniqueness so no two entities can have the same
 // ID, even if they are of different types. It also enforces referential integrity so construction of an entity that
 // references another (e.g. a database entity references a client) will fail if the referenced entity does not exist.
-type entityMap struct {
-	allEntities    map[string]struct{}
-	changeStreams  map[string]*mongo.ChangeStream
-	clientEntities map[string]*clientEntity
-	dbEntites      map[string]*mongo.Database
-	collEntities   map[string]*mongo.Collection
-	sessions       map[string]mongo.Session
-	gridfsBuckets  map[string]*gridfs.Bucket
-	bsonValues     map[string]bson.RawValue
-}
-
-func newEntityMap() *entityMap {
-	return &entityMap{
-		allEntities:    make(map[string]struct{}),
-		gridfsBuckets:  make(map[string]*gridfs.Bucket),
-		bsonValues:     make(map[string]bson.RawValue),
-		changeStreams:  make(map[string]*mongo.ChangeStream),
-		clientEntities: make(map[string]*clientEntity),
-		collEntities:   make(map[string]*mongo.Collection),
-		dbEntites:      make(map[string]*mongo.Database),
-		sessions:       make(map[string]mongo.Session),
-	}
-}
-
-func (em *entityMap) addBSONEntity(id string, val bson.RawValue) error {
+type EntityMap struct {
```

Comment:
It may be confusing for a future reader that some of the accessors like `GetEventList` are exported while others like `session` are not. I'm guessing this is because the exported ones need to be accessed from the Go executor in the drivers-atlas-testing project. We should either document this or make all of them exported.